File: util/android/src/main/java/com/dmi/util/opengl/GLRendererDelegate.java
package com.dmi.util.opengl;

public class GLRendererDelegate implements GLRenderer {
    private final GLRenderer renderer;

    public GLRendererDelegate(GLRenderer renderer) {
        this.renderer = renderer;
    }

    @Override
    public void onSurfaceCreated() {
        renderer.onSurfaceCreated();
    }

    @Override
    public void onSurfaceChanged(int width, int height) {
        renderer.onSurfaceChanged(width, height);
    }

    @Override
    public void onFreeResources() {
        renderer.onFreeResources();
    }

    @Override
    public void onDrawFrame() {
        renderer.onDrawFrame();
    }
}

File: util/util-android/src/main/kotlin/com/dmi/util/opengl/GLRendererDelegate.kt
package com.dmi.util.opengl

class GLRendererDelegate(private val renderer: GLRenderer) : GLRenderer {

    override fun onSurfaceCreated() {
        renderer.onSurfaceCreated()
    }

    override fun onSurfaceChanged(width: Int, height: Int) {
        renderer.onSurfaceChanged(width, height)
    }

    override fun onFreeResources() {
        renderer.onFreeResources()
    }

    override fun onDrawFrame() {
        renderer.onDrawFrame()
    }
}
